blog-posts/github-billing-dashboard.md

```diff
@@ -0,0 +1,169 @@
+
+This is all the billing information you get from the Github Settings. No historical data, no expense per service.
+
+![screenshot of the Github billing dashboard with some charts](./images/screenshot-of-gh-dashboard.png)
+
+And this is what you get from the Github Billing Dashboard. Costs per service per month with historic data.
+
+This blog post gives you some insights into how we build it.
+
+## Technologies and Libraries
+
+The dashboard runs locally in the user's browser and thus does not send any data to external servers, even the most recently used files are stored only in the browser's local storage. The dashboard was mainly developed with React and Typescript. For the styling of the dashboard, we used styled-components. In addition, various libraries were used.
+
+### papaparse
+
+[papaparse](https://github.com/mholt/PapaParse) is a CSV parser that converts CSV to JSON. Besides strings, papaparse can process CSV files directly, no matter if they are located locally on the computer or have to be accessed via an external URL.
+
+The following example shows how easy the conversion of a CSV file is. The function parse() is called with a file like sample.csv. Additionally, it needs as a second input parameter a config object, in which certain options for the parsing can be set. For example, the headings of the individual values are camalized here. The parsing itself, the assignment of the values to the headings, and the transformation into the UsageReportCsvEntry type are completely handled by papaparse. The parsed object can still be modified. In the sample code, you can see how in addition to the existing CSV file headers "pricePerUnit" is calculated and added to the finished object.
+
+```
+billing.csv
+
+Date,Product,Repository Slug,Quantity,Unit Type,Price Per Unit,Actions Workflow,Notes
+2021-06-14,actions,sample-repository1,21,UBUNTU,$0.008,.github/workflows/integration-test.yml,
+2021-06-14,shared storage,sample-repository2,20.1028,gb,$0.25,,
+2021-06-15,actions,sample-repository3,26,UBUNTU,$0.008,.github/workflows/main.yml,
+2021-06-15,actions,sample-repository2,17,UBUNTU,$0.008,.github/workflows/integration-test.yml,
+```
+
+```tsx
+export interface UsageReportCsvEntry {
+	date: string;
+	product: string;
+	repositorySlug: string;
+	quantity: string;
+	unitType: string;
+	pricePerUnit: string;
+	actionsWorkflow: string;
+	notes: string;
+}
+
+export interface UsageReportEntry extends UsageReportCsvEntry {
+	totalPrice: number;
+}
+
+parse<UsageReportCsvEntry>(file, {
+	header: true,
+	skipEmptyLines: true,
+	transformHeader: (header: string): string => {
+		return camalize(header);
+	},
+	complete: (result) => {
+			const githubBillingEntries: UsageReportEntry[] = result.data.map(
+				(dailyEntry) => {
+					return {
+					...dailyEntry,
+					totalPrice: *parseFloat*(dailyEntry.quantity) * *parseFloat*(dailyEntry.pricePerUnit),
+				};
+			}
+		);
+	},
+});
+```
+
+### Recharts
```

Comment:
You linked papaparse but you don't link recharts. I would love to have the github link here to quickly check the sources/repo.

Reply:
I have seen the fix👍